plugins/mission/mission_item.cpp

```diff
@@ -117,4 +131,59 @@ std::string MissionItem::to_str(MissionItem::CameraAction camera_action)
+           && (lhs.get_longitude_deg() == rhs.get_longitude_deg()
+               || isnan(lhs.get_longitude_deg()) && isnan(rhs.get_longitude_deg()))
+           && (lhs.get_relative_altitude_m() == rhs.get_relative_altitude_m()
+               || isnan(lhs.get_relative_altitude_m()) && isnan(rhs.get_relative_altitude_m()))
+           && lhs.get_fly_through() == rhs.get_fly_through()
+           && (lhs.get_speed_m_s() == rhs.get_speed_m_s()
+               || isnan(lhs.get_speed_m_s()) && isnan(rhs.get_speed_m_s()))
+           && (lhs.get_gimbal_pitch_deg() == rhs.get_gimbal_pitch_deg()
+               || isnan(lhs.get_gimbal_pitch_deg()) && isnan(rhs.get_gimbal_pitch_deg()))
+           && (lhs.get_gimbal_yaw_deg() == rhs.get_gimbal_yaw_deg()
+               || isnan(lhs.get_gimbal_yaw_deg()) && isnan(rhs.get_gimbal_yaw_deg()))
+           && (lhs.get_loiter_time_s() == rhs.get_loiter_time_s()
+               || isnan(lhs.get_loiter_time_s()) && isnan(rhs.get_loiter_time_s()))
+           && lhs.get_camera_action() == rhs.get_camera_action()
+           && lhs.get_camera_photo_interval_s() == rhs.get_camera_photo_interval_s();
+}
+
+std::ostream &operator<<(std::ostream &str, MissionItem const &mission_item)
+{
+    return str << std::endl << "[" << std::endl << std::setprecision(10)
+           << "\tlat: " << mission_item.get_latitude_deg() << std::endl
+           << "\tlon: " << mission_item.get_longitude_deg() << std::endl
+           << "\trel_alt: " << mission_item.get_relative_altitude_m() << std::endl
+           << "\tis_fly_through: " << (mission_item.get_fly_through() ? "true" : "false") << std::endl
+           << "\tspeed: " << mission_item.get_speed_m_s() << std::endl
+           << "\tgimbal_pitch: " << mission_item.get_gimbal_pitch_deg() << std::endl
+           << "\tgimbal_yaw: " << mission_item.get_gimbal_yaw_deg() << std::endl
+           << "\tloiter_time: " << mission_item.get_loiter_time_s() << std::endl
+           << "\tcamera_action: " << mission_item.get_camera_action() << std::endl
+           << "\tcamera_photo_interval: " << mission_item.get_camera_photo_interval_s()
+           << std::endl << "]" << std::endl;
+}
+
+std::ostream &operator<<(std::ostream &str, MissionItem::CameraAction const &camera_action)
+{
+    switch (camera_action) {
+        case MissionItem::CameraAction::TAKE_PHOTO:
+            return str << "CameraAction::TAKE_PHOTO";
+        case MissionItem::CameraAction::START_PHOTO_INTERVAL:
+            return str << "CameraAction::START_PHOTO_INTERVAL";
+        case MissionItem::CameraAction::STOP_PHOTO_INTERVAL:
+            return str << "CameraAction::STOP_PHOTO_INTERVAL";
+        case MissionItem::CameraAction::START_VIDEO:
+            return str << "CameraAction::START_VIDEO";
+        case MissionItem::CameraAction::STOP_VIDEO:
+            return str << "CameraAction::STOP_VIDEO";
+        case MissionItem::CameraAction::NONE:
+            return str << "CameraAction::NONE";
```

Comment:
We should probably do the same kind of thing instead of the `result_str()` methods.